Pp 9940 build carbon relay ng

jfharden opened the pull request:
This PR adds a build pipeline for the base carbon-relay-ng image.

Efforts are taken to make it as easy as possible to stay up to date with upstream. We are parsing the golang version used to build from the grafana/carbon-relay-ng circleci config.

Despite having already been tested by github this pipeline also runs the tests against the final built container. Considering we are pushing this to a public repository it seems prudent to take the extra care given the container build runs apk upgrade.

A successful run of this pipeline can be seen on concourse https://pay-cd.deploy.payments.service.gov.uk/teams/pay-dev/pipelines/carbon-relay-ng/jobs/build-then-test-and-push/builds/4

See the successfully built images in dockerhub: https://hub.docker.com/repository/docker/govukpay/carbon-relay-ng/general (currently private, but we should make this public once we are happy with the build process)

And in ecr in the `govukpay/carbon-relay-ng` repository in the test account

katstevens commented on ci/pipelines/carbon-relay-ng.yml:
```diff
@@ -0,0 +1,243 @@
+
+                echo "Running tests"
+                docker-compose up --exit-code-from test-receiver
+                echo "Tests passed"
```
Do we want any Slack notifications if the jobs fail?

jfharden replied:
Oh yeah, I think we definitely do

jfharden replied:
I'll set the usual ones up for pass/fail

jfharden commented:
Added pass/fail notifications.

Pass notification:
<img width="556" alt="Screenshot 2022-08-22 at 16 48 44" src="https://user-images.githubusercontent.com/2170030/185963938-f8119dfa-80bf-427c-a876-bd538c3bcd2b.png">

I added an exit 1 to one of the tasks to make it fail. The failure:
<img width="535" alt="Screenshot 2022-08-22 at 16 51 06" src="https://user-images.githubusercontent.com/2170030/185964240-f36335c9-fb35-4f7f-a929-38c087aee79d.png">
